### DISCUSSION TITLE
Multiple Preloaders for each Renderers

### DISCUSSION THREAD

**Discussion by AlessioBottin** [cawa-93/vite-electron-builder#838] [Q&A]
Hi, I am sorry this the second time i ask for help in a few days, but this is my first time using Electron, Vite and Typescript(I already know js) so it has been a bit challenging trying to figure out the optimal place where to write code and how to make changes at the project structure.
I wanted to use a different preloader for each renderer i have, the problem is: How do i configure vite config in such a way that it can take multiple entry points and make an out file for each one of them?
On Vite documentation i have found that build.lib.entry accepts only one argument and it must be a string, so I am wondering: is this is achievable or is there a better approach?

**Comment by cawa-93** [comment]
Although vite supports [multiple entry HTML files](https://vitejs.dev/guide/build.html#multi-page-app), it does not support multiple entry points for library compilation. What can solve your problem is creating a copy of an existing preload package and building it as an independent package. However, I don't understand situations where different renderers might need different preloader contexts

**Reply by AlessioBottin** [reply]
I wanted to use different preload scripts because there might be situations where I do not want to pass values to every renderer but just to the one where I am using it.
Isn't using the same script for every renderer going to run all the code (even the one i don't need) every time I attach it to a window?

**Reply by cawa-93** [reply]
> Isn't using the same script for every renderer going to run all the code (even the one i don't need) every time I attach it to a window?

It will.
